test/JsonApiDotNetCoreMongoDbExampleTests/IntegrationTests/Sorting/SortTests.cs

```diff
@@ -0,0 +1,219 @@
+                new Person { Id = ObjectId.GenerateNewId().ToString() },
+                new Person { Id = ObjectId.GenerateNewId().ToString() }
+            };
+
+            await _testContext.RunOnDatabaseAsync(async db =>
+            {
+                await db.ClearCollectionAsync<Person>();
+                await db.GetCollection<Person>().InsertManyAsync(persons);
+            });
+
+            var route = "/api/v1/people";
+
+            // Act
+            var (httpResponse, responseDocument) = await _testContext.ExecuteGetAsync<Document>(route);
+
+            // Assert
+            httpResponse.Should().HaveStatusCode(HttpStatusCode.OK);
+            
+            responseDocument.ManyData.Should().HaveCount(4);
+            responseDocument.ManyData[0].Id.Should().Be(persons[0].StringId);
+            responseDocument.ManyData[1].Id.Should().Be(persons[1].StringId);
+            responseDocument.ManyData[2].Id.Should().Be(persons[2].StringId);
+            responseDocument.ManyData[3].Id.Should().Be(persons[3].StringId);
+        }
+        
+        [Fact]
+        public async Task Cannot_sort_on_HasMany_relationship()
+        {
+            // Arrange
+            var blogs = new List<Blog>
+            {
+                new Blog
+                {
+                    Articles = new List<Article>
+                    {
+                        new Article
+                        {
+                            Caption = "A"
+                        },
+                        new Article
+                        {
+                            Caption = "B"
+                        }
+                    }
+                },
+                new Blog
+                {
+                    Articles = new List<Article>
+                    {
+                        new Article
+                        {
+                            Caption = "C"
+                        }
+                    }
+                }
+            };
+
+            await _testContext.RunOnDatabaseAsync(async db =>
+            {
+                await db.ClearCollectionAsync<Blog>();
+                await db.GetCollection<Blog>().InsertManyAsync(blogs);
+            });
+            
+            var route = "/api/v1/blogs?sort=count(articles)";
+
+            // Act
+            var (httpResponse, responseDocument) = await _testContext.ExecuteGetAsync<ErrorDocument>(route);
+
+            // Assert
+            httpResponse.Should().HaveStatusCode(HttpStatusCode.BadRequest);
+            
+            responseDocument.Errors.Should().HaveCount(1);
+            responseDocument.Errors[0].StatusCode.Should().Be(HttpStatusCode.BadRequest);
+            responseDocument.Errors[0].Title.Should().Be("Relationships are not supported when using MongoDB.");
+            responseDocument.Errors[0].Detail.Should().BeNull();
+        }
+        
+        [Fact]
+        public async Task Cannot_sort_on_HasManyThrough_relationship()
```

Comment:
> I had copied these tests without the complete Arrange part, I mean, I had not added resources to the database prior to calling the endpoints. This worked for HasOne and HasManyThrough but it did not work for HasMany. Adding this part of the test made HasMany work too. Do you think we should do something about this?
I mean, should we expect an API to return 400 when the database is empty and we try to use relationships? I think we should but I don't know how I can do that neatly hehe

In theory we should, however that involves complexity which I believe is not worth the effort (plus extra setup burden for end-users). The reason some fail and other don't worries me. I suspect it has to do with the order of tests execution. So a previous test inserted a record that is reused. To verify, clear the whole collection upfront. I believe it's best to insert a record upfront for all three relationship tests in order to get stable results. We don't want tests to break when they are reordered or a new one is inserted.

Reply:
Agreed but wait, you mean some of these tests are failing on your end?